test/unit/math/prim/mat/prob/vector_rng_test_helper.hpp

```diff
@@ -69,37 +148,38 @@ void setValue(T1& v, double value) {
  * @param value Value to assign
  */
 template<>
-void setValue(double& v, double value) {
+void set_value(double& v, double value) {
   v = value;
 }
 
 /*
- * Return true if variables defined with constraint c1 do not violate constraint
- * c2
+ * Check that every element of p (if p is a vector) passes check. If p is a
+ * scalar, then just make sure it passes the check.
  *
- * @param c1 First constraint
- * @param c2 Second constraint
- * @return True if values with constraint c1 and compatible with c2
+ * @tparam T_check Callable type that takes in one double and returns a bool
+ * @tparam T_param Any type compatible with stan::scalar_seq_view
+ * @param check Callable for validating (values of) p
+ * @param p Parameter to check
  */
-bool compatible(Constraint c1, Constraint c2) {
-  if (c1 == Constraint::NonNegative && c2 == Constraint::Positive)
-    return false;
-
-  if (c1 == Constraint::None && c2 == Constraint::NonNegative)
-    return false;
+template<typename T_check, typename T_param>
+bool VectorizeCheck(T_check check, T_param p) {
```

Comment:
This function is typically called something like [every?](https://clojuredocs.org/clojure.core/every_q), [all](http://hackage.haskell.org/package/base-4.10.0.0/docs/Prelude.html#v:all), or [forall](http://www.scala-lang.org/api/2.12.1/scala/collection/immutable/List.html#forall(p:A=>Boolean):Boolean). I'd re-use an existing name for it (whichever is fine), or maybe try to use [something like this](http://www.cplusplus.com/reference/algorithm/all_of/) in the stdlib, though that's unaware of things like `scalar_seq_view`.